### PR TITLE
Perf dialect runtime

### PR DESCRIPTION
**Pull request by adam-smnk**
Adds runtime bindings required to execute perf dialect operations.
The existing manual MLIR benchmarking loop is now moved to rely on the perf dialect ops.
Also, necessary lowering passes are added to the `tpp-run`.

Work toward #100

### CODE REVIEW

File: tpp-rt/PerfRunnerUtils.cpp [hunk 1]
```diff
@@ -29,25 +33,73 @@ int64_t _mlir_ciface_timer_alloc() {
...
+// A generic sink function.
+// Its aim is to ensure that the passed data and its producers cannot be
+// optimized away such that the time measured by a benchmark loop correctly
+// represents the full workload.
+static void __attribute__((optnone)) perf_sink(void *data) { (void)data; }
+
+/*
+  Perf dialect runtime bindings for common perf.sink op argument types.
+*/
+void _mlir_ciface_perf_sink_memref_i8(UnrankedMemRefType<int8_t> *val) {
```
Comment (rengolin) [inline tpp-rt/PerfRunnerUtils.cpp:86]: I don't think we need runtime support for this. Declaring the functions at the IR level and marking as opt-none / no-inline may be enough to get all compile time types needed.

**Review by rengolin — COMMENTED** [review]
Looks much cleaner, thanks!

Some comments inline.

File: tpp-rt/PerfRunnerUtils.cpp (hunk 1 above)
Comment (adam-smnk) [reply tpp-rt/PerfRunnerUtils.cpp]: I wasn't sure how to make it work directly in MLIR before but I finally figured it out. It's much cleaner now.

**Comment by adam-smnk** [comment]
`perf.sink` functionality was verified with a small manual diff: 
[perf_sink_present.patch](https://github.com/plaidml/tpp-mlir/files/10279603/perf_sink_present.patch)
[perf_sink_absent.patch](https://github.com/plaidml/tpp-mlir/files/10279602/perf_sink_absent.patch)

Results:
- present:
```Benchmark: mlp
ref                 :     1.596 +-     0.094 gflops
mlir                :   132.403 +-    20.538 gflops
```
- absent:
```Benchmark: mlp
ref                 :     1.656 +-     0.132 gflops
mlir                : 1415357.311 +- 1256694.567 gflops
```

File: include/Utils/Perf.h [hunk 2]
```diff
@@ -43,7 +43,7 @@ class PerfResults {
   }
 
   /// Zero a time_point
-  void zero(std::chrono::high_resolution_clock::time_point& point) {
+  void zero(std::chrono::high_resolution_clock::time_point &point) {
```
Comment (rengolin) [inline include/Utils/Perf.h:46]: Unnecessary change for this PR

**Review by rengolin — APPROVED** [review]
Awesome! Just one nit on the spurious change and LGTM, thanks!

This closes the issue, right?

**Comment by adam-smnk** [comment]
> This closes the issue, right?

Yup, I think that's it.
